app/src/main/resources/ui/shared.js

```diff
@@ -724,34 +729,61 @@ export function createIconWithConnectionTooltip(dataConnectionSelect) {
     iconDiv.setAttribute("data-bs-html", "true");
     iconDiv.setAttribute("data-bs-title", "Connection options");
     new bootstrap.Tooltip(iconDiv);
-    // on select change, update icon title
-    dataConnectionSelect.addEventListener("change", (event) => {
-        let connectionName = event.target.value;
-        fetch(`http://localhost:9898/connection/${connectionName}`, {method: "GET"})
-            .then(r => {
-                if (r.ok) {
-                    return r.json();
-                } else {
-                    r.text().then(text => {
-                        createToast(`Get connection ${connectionName}`, `Failed to get connection ${connectionName}! Error: ${err}`, "fail");
-                        throw new Error(text);
-                    });
+    return iconDiv;
+}
+
+export function addConnectionOverrideOptions(connectionName, iconDiv, overrideOptionsContainer, propertyClass, index) {
+    fetch(`http://localhost:9898/connection/${connectionName}`, {method: "GET"})
+        .then(r => {
+            if (r.ok) {
+                return r.json();
+            } else {
+                r.text().then(text => {
+                    createToast(`Get connection ${connectionName}`, `Failed to get connection ${connectionName}! Error: ${err}`, "fail");
+                    throw new Error(text);
+                });
+            }
+        })
+        .then(respJson => {
+            if (respJson) {
+                let optionsToShow = {};
+                optionsToShow["type"] = respJson.type;
+                for (let [key, value] of Object.entries(respJson.options)) {
+                    if (key !== "user" && key !== "password") {
```

Comment:
Is there a better way to decide what keys to not show? Can we refer to the configuration-data